packages/config/src/Config.ts

```diff
@@ -17,46 +17,88 @@ import * as findUp from "find-up";
 import * as JSONC from "comment-json";
 import * as lodash from "lodash";
 
+import { ConfigConstants } from "./ConfigConstants";
 import { IConfig } from "./doc/IConfig";
 import { IConfigLayer } from "./doc/IConfigLayer";
 import { ImperativeError } from "../../error";
 import { IConfigProfile } from "./doc/IConfigProfile";
 import { IConfigOpts } from "./doc/IConfigOpts";
-import { IConfigSecure, IConfigSecureProperties } from "./doc/IConfigSecure";
+import { IConfigSecure } from "./doc/IConfigSecure";
 import { IConfigVault } from "./doc/IConfigVault";
-import { IConfigLoadedProfile, IConfigLoadedProperty } from "./doc/IConfigLoadedProfile";
-
-enum layers {
-    project_user = 0,
-    project_config,
-    global_user,
-    global_config
-}
+import { ConfigLayers, ConfigPlugins, ConfigProfiles, ConfigSecure } from "./api";
+
+/**
+ * Enum used by Config class to maintain order of config layers
+ */
+enum Layers {
+    ProjectUser = 0,
+    ProjectConfig,
+    GlobalUser,
+    GlobalConfig
+};
 
 export class Config {
-    private static readonly SECURE_ACCT = "secure_config_props";
+    /**
+     * The trailing portion of a shared config file name
+     */
     private static readonly END_OF_TEAM_CONFIG = ".config.json";
+
+    /**
+     * The trailing portion of a user-specific config file name
+     */
     private static readonly END_OF_USER_CONFIG = ".config.user.json";
 
-    private _app: string;
-    private _paths: string[];
-    private _layers: IConfigLayer[];
-    private _home: string;
-    private _name: string;
-    private _user: string;
-    private _schema: string;
-    private _active: {
+    /**
+     * App name used in config filenames (e.g., *my_cli*.config.json)
+     * @internal
+     */
+    public _app: string;
+
+    /**
+     * List to store each of the config layers enumerated in `layers` enum
+     * @internal
+     */
+    public _layers: IConfigLayer[];
```

Comment:
right or wrong, most of our member variables in the code base start with `m`.  I know other standards recommend an `_` prefix.  

given that most of the code base is `m`, should this be used here, if at minimum, for aesthetics ?

Reply:
I am quite willing to make this change.

I do think that we should establish what our standards are before people write code.

I do not feel strongly about this particular naming, since I did not create these variable names.

Reply:
Perhaps others have a stronger opinion - @zFernand0 & @t1m0thyj?

Ideally our standards would be enforced in lint rules.  Usually most seem to pattern their code with the rest of the code base, but, that's not universally agreed upon either 😄 .

Reply:
No strong opinions here.
I say let's pick one and open an issue to enforce the rule for internal variables (public or private). 😋

Reply:
`m` is most common and would require less changes over all than adopting a `_` standard.

Reply:
No strong opinions here either.
Whichever we pick, perhaps we should investigate whether we can set up a linting rule to enforce it? I guess that's the same as what @zFernand0 is suggesting 😋

Reply:
I agree that we should change this subset of files, instead of changing all other files.

I will undertake the change from _var to mVar now.